byron/ledger/executable-spec/test/Ledger/Pvbump/Properties.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Ledger.Pvbump.Properties where
+
+import           Control.State.Transition (applySTS, TRC(..))
+import           Generator
+import           Hedgehog
+import           Ledger.Update (PVBUMP)
+
+
+-- Property #1 for the PVBUMP STS
+-- 
+-- Given an empty list of mappings from a slot to a pair of a protocol
+-- version and protocol parameters (this is the fads part of the
+-- environment), the PVBUMP STS results in the same state it started
+-- in.
+emptyPVUpdate :: Property
+emptyPVUpdate = property $ do
+  PVBUMPEnv env  <- forAll pvbumpEmptyListEnvGen
+  PVBUMPState st <- forAll pvbumpStateGen
+  sig            <- forAll pvbumpSignalGen
```

Comment:
To improve shrinking it would be better to use `Applicative` here, so
```
judgementContext <- forAll $ fmap TRC $ (,,) <$> pvbumpEmptyListEnvGen <*> pvbumpStateGen <*> pvbumpSignalGen
```

Reply:
I agree with the general idea, but when I do that then many `Show` instances are needed, in particular for `TRC` and all environments, states, and signals. Does it sound reasonable to add the instances?

Reply:
Fixed!